crud/common/utils.lua

```diff
@@ -524,4 +581,14 @@ function utils.flatten_obj_reload(space_name, obj)
     return schema.wrap_func_reload(flatten_obj, space_name, obj)
 end
 
+function utils.get_index_fieldno_map(index_obj)
+    local t = {}
+    for i, part in index_obj.parts do
```

Comment:
`in ipairs(index_obj.parts)`? How did it pass tests?

Reply:
Seems my tests use branches where bucket_id is calculated using the primary index, not DDL sharding key.
`ipairs` added, but I think we need to use something like an error injection mechanism to test certain branches/conditions in code.